File: domain/src/main/java/home/oleg/placenearme/repositories/DetailedVenueRepository.java
package home.oleg.placenearme.repositories;

import home.oleg.placenearme.models.DetailedVenue;
import io.reactivex.Flowable;
import io.reactivex.Single;

public interface DetailedVenueRepository {

    Flowable<DetailedVenue> getDetailedVenueById(String venueId);

    Flowable<DetailedVenue> stream(String venueId);

    Single<DetailedVenue> fetch(String venueId);

}

File: core-domain/src/main/java/home/oleg/placenearme/repositories/DetailedVenueRepository.kt
package home.oleg.placenearme.repositories

import home.oleg.placenearme.models.DetailedVenue
import io.reactivex.Flowable
import io.reactivex.Single

interface DetailedVenueRepository {

    fun getDetailedVenueById(venueId: String): Flowable<DetailedVenue>

    fun stream(venueId: String): Flowable<DetailedVenue>

    fun fetch(venueId: String): Single<DetailedVenue>

}
